chapters/os_tooling.md

```diff
@@ -0,0 +1,85 @@
+# Offline Compilation of OpenCL Kernel Sources
+
+Aside from online compilation during the application execution, OpenCL kernel sources can be compiled offline into binaries that can be loaded into the drivers using special API (e.g. `clCreateProgramWithBinary` or `clCreateProgramWithIL`).
+
+This section described available open source tools for offline compilation of OpenCL kernels.
+
+## Open Source Tools
+
+* [clang](https://clang.llvm.org/) is a compiler front-end for the C/C++ family of languages including OpenCL C and C++ for OpenCL that can compile into an executable binary (e.g. AMDGPU), or a portable binary (e.g. SPIR). It is part of [the LLVM compiler infrastructure project](https://llvm.org/), and there is information regarding [OpenCL kernel language support and standard headers](https://clang.llvm.org/docs/UsersManual.html#opencl-features).
+* [libclc](https://libclc.llvm.org/) is a generic and portable implementation of OpenCL builtin function libraries for OpenCL 1.1 - and some functions from later versions of OpenCL can be found there too.
```

Comment:
libclc feels like an odd one out in the list.  All the others are something that is useful to an end-user programmer, libclc is useful for tool implementers.   Maybe just drop it from this list?

Reply:
The only relevant use case I could think of is clspv.

But perhaps you are right it would be most likely hidden behind the compiler driver setup.

I am looping in @alan-baker in case he has some more input.

Otherwise, we can remove it indeed.

Reply:
My thought was: where is libclc in the image? I would either add it as a box and hook it up to the linking part of both sides (to communicate that implementations of built-in have to be provided by some entity and don't materialize from thin air), or either drop it from the text.

It's either important enough for people to get the "whole picture" or not, but not both.

Reply:
I think we should try to figure out how to make libclc more useful and accessible, but I agree that it doesn't add much to this list right now.

Reply:
I have removed it for now. This can be added any time we find a good use case for the end-user.